backend/app/auth/service.py

```
@@ -127,17 +137,31 @@ async def authenticate_user_with_email(self, email: str, password: str) -> Dict[
         """
         db = self.get_db()
-        
-        user = await db.users.find_one({"email": email})
+        try:
+            user = await db.users.find_one({"email": email})
+        except PyMongoError as e:
+            logger.error(f"Database error during user lookup: {e}")
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Internal server error"
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)